app/templates/models_upload_template.rb

```diff
@@ -1,13 +1,53 @@
-class ModelsUploadTemplate < CsvVerticalUploadTemplate
-  def headers
-    super + ['Model 1']
+class ModelsUploadTemplate
+  attr_accessor :user
+
+  def export
+    CSV.generate do |csv|
+      csv << headers
+      ModelsData::PROPERTY_NAMES.each do |property_name|
+        attr_info = Model.attribute_info(property_name)
+
+        csv << [
+          attr_info.category,
+          I18n.t(Model.key_for_name(property_name)),
+          I18n.t(Model.key_for_definition(property_name)),
+          attr_info.input_type_for_display,
+          attr_info.options.try(:join, '; ')
+        ] + models_attributes(attr_info.name)
+      end
+    end
   end
 
-  def property_names
-    ModelsData::PROPERTY_NAMES
+  private
+
+  def models
+    Pundit.policy_scope(user, Model)
+  end
+
+  def headers
+    base_headers = [
+      'Category', 'Indicator', 'Definition/Input Explanation', 'Data Type',
+      'Picklist options (Data Entry)', 'Model 1'
+    ]
+
+    models_headers =
+      if models.present?
+        models.map.with_index(1) { |_, i| "Model #{i}" }
+      else
+        ['Model 1']
```

Comment:
I just tried downloading and it works but for Model's it shows "Model 1" twice, as I already have a model, and then the first empty column shows up with the same header.

Reply:
Confirm.

It should be fixed now, thanks for checking.